src/Fixer/ArrayNotation/YieldFromArrayToYieldsFixer.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\ArrayNotation;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Kuba Werłos <werlos@gmail.com>
+ */
+final class YieldFromArrayToYieldsFixer extends AbstractFixer
+{
+    /**
+     * @var array<int, Token>
+     */
+    private array $inserts;
+
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        return new FixerDefinition(
+            'Yield from array should be unpacked to series of yields.',
+            [new CodeSample('<?php function generate() {
+    yield from [
+        1,
+        2,
+        3,
+    ];
+}
+')]
+        );
+    }
+
+    public function isCandidate(Tokens $tokens): bool
+    {
+        return $tokens->isTokenKindFound(T_YIELD_FROM);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * Must run before BlankLineBeforeStatementFixer, NoExtraBlankLinesFixer, NoMultipleStatementsPerLineFixer, NoWhitespaceInBlankLineFixer, StatementIndentationFixer.
+     */
+    public function getPriority(): int
+    {
+        return 0;
+    }
+
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
+    {
+        $this->inserts = [];
+
+        for ($index = $tokens->count() - 1; $index >= 0; --$index) {
+            if (!$tokens[$index]->isGivenKind(T_YIELD_FROM)) {
+                continue;
+            }
+
+            $arrayStartIndex = $tokens->getNextMeaningfulToken($index);
+
+            if (!$tokens[$arrayStartIndex]->isGivenKind([T_ARRAY, CT::T_ARRAY_SQUARE_BRACE_OPEN])) {
+                continue;
+            }
+
+            $tokens->clearTokenAndMergeSurroundingWhitespace($index);
+
+            if ($tokens[$arrayStartIndex]->isGivenKind(T_ARRAY)) {
+                $startIndex = $tokens->getNextTokenOfKind($arrayStartIndex, ['(']);
+                $tokens->clearAt($arrayStartIndex);
```

Comment:
I found small bug for this line, fix (and few other changes) in https://github.com/6b7562617765726c6f73/PHP-CS-Fixer/pull/1/commits/26de244b9c8176ccf8885ca2040b348e7410b251